src/bruteforce/bruteforce-omp_ma.c

```diff
@@ -0,0 +1,74 @@
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <omp.h>
+#include "../hash/hash.h"
+#include "bruteforce-util.c"
+
+int bruteforce_crack(char *password_hash, char *characters, int password_max_length, int verbose)
+{
+    int number_of_characters = strlen(characters);
+    printf("Brute force of hash: %s\n", password_hash);
+    printf("Using %d characters: %s\n", number_of_characters, characters);
+    printf("Calculating to a length of %d\n", password_max_length);
+
+    int result = 1;
+
+    static unsigned char buffer[65];
+    int i, j, k;
+
+    for (i = 1; i <= password_max_length; i++)
+    {
+        long possibilities = (long)pow(number_of_characters, i);
+        if (verbose)
+        {
+            printf("Now calculating password length of %d, it has %ld possibilities\n", i, possibilities);
+        }
+
+#pragma omp parallel private(buffer, j, k) shared(result)
+        {
+            char passwordToTest[i + 1];
+            for (j = 0; j < possibilities;)
+            {
+                if (result == 0)
+                {
+                    printf("Found pass already...\n");
+                    //return; //Skip the next processes if we have found the result
+                }
+                int nextStep = j + 100000;
```

Comment:
Might want to update this in MPI, but have nextStep be the size of the remaining work if less than 100000.
Also in both programs make 100000 a constant we can update with a #define CHUNK_SIZE or something
```
int nextStep;
if (result - j < CHUNK_SIZE) 
{
  nextStep = result - j;
}
else 
{
nextStep = j + CHUNK_SIZE
}
```